tests/system/large/test_session.py

```diff
@@ -46,7 +50,72 @@
         # ),
     ],
 )
-def test_read_gbq_for_large_tables(session: Session, query_or_table, index_col):
+def test_read_gbq_for_large_tables(
+    session: bigframes.Session, query_or_table, index_col
+):
     """Verify read_gbq() is able to read large tables."""
     df = session.read_gbq(query_or_table, index_col=index_col)
     assert len(df.columns) != 0
+
+
+def test_close(session):
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    full_id_1 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    full_id_2 = bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.Conflict):
+        bqclient.create_table(full_id_2)
+
+    session.close()
+
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_1)
+    with pytest.raises(google.cloud.exceptions.NotFound):
+        bqclient.delete_table(full_id_2)
+
+
+def test_clean_up_by_session_id():
+    original_location = bigframes.options.bigquery.location
+    bpd.close_session()  # close the session to change regions
+    bigframes.options.bigquery.location = "europe-west10"
+    # we do this test in a different region in order to avoid
+    # overly large amounts of temp tables slowing the test down
+
+    session = bpd.get_global_session()
+    session_id = session.session_id
+
+    # we will create two tables and confirm that they are deleted
+    # when the session is closed
+
+    bqclient = session.bqclient
+    dataset = session._anonymous_dataset
+    expiration = (
+        datetime.datetime.now(datetime.timezone.utc)
+        + bigframes.constants.DEFAULT_EXPIRATION
+    )
+    bigframes.session._io.bigquery.create_temp_table(session, expiration)
+    bigframes.session._io.bigquery.create_temp_table(session, expiration)
+
+    tables_before = bqclient.list_tables(dataset, max_results=10000, page_size=10000)
+    tables_before_count = len(list(tables_before))
+    assert tables_before_count >= 2
+
+    bpd.clean_up_by_session_id(
+        session_id, location=session._location, project=session._project
+    )
+
+    tables_after = bqclient.list_tables(dataset, max_results=10000, page_size=10000)
+    assert len(list(tables_after)) <= tables_before_count - 2
```

Comment:
This is going to be a pretty flaky test, I think. We run tests in parallel, so there might be other tests that create some tables in the same dataset.

In fact, this test might run at the same time across Python versions if/when we split our system tests up into multiple Kokoro jobs.

Instead, let's just check:

* In `tables_before` that there are at least some tables with the `session_id` as a prefix.
* In `tables_after` that there are no tables with the `session_id` as a prefix.

Even with this change, it could get flaky when there are more than 10000 tables, but at least it'll be a bit more robust.

Reply:
Roger, changed to `assert any([(session.session_id in table.full_table_id) for table in list(tables_before)])`